registry/storage/registry.go

```diff
@@ -154,6 +171,20 @@ func (reg *registry) BlobStatter() distribution.BlobStatter {
 	return reg.statter
 }
 
+// ForeignLayerHostWhitelist tests whether the host is allowed in foreign layer URLs.
+func (reg *registry) ForeignLayerHostAllowed(host string) bool {
+	if reg.foreignLayerHostWhitelist == nil {
```

Comment:
the knob I'm talking about can cleanly take care of this check also - instead of checking for the presence of a whitelist, we can check if foreign layers are enabled at all and exit early here